lib/core/client.js

```diff
@@ -376,9 +375,37 @@ class Parser extends HTTPParser {
     this.shouldKeepAlive = false
     this.read = 0
     this.request = null
+    this.paused = false
+    this.queue = []
```

Comment:
We can't pause the parser from within a callback so we need to implement a buffer in JS land in order to stop further progress while paused.

Reply:
Can you add a comment so it's clear to future readers?